lib/ansible/modules/cloud/cloudstack/cs_physical_network.py

```diff
@@ -0,0 +1,464 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2017, Netservers Ltd. <support@netservers.co.uk>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cs_physical_network
+short_description: Manages physical networks on Apache CloudStack based clouds.
+description:
+    - Create, update and remove networks.
+    - Enabled and disabled Network Service Providers
+    - Enables Internal LoadBalancer and VPC/VirtualRouter elements as required
+version_added: "2.8"
+author:
+ - Netservers Ltd. (@netservers)
+ - Patryk Cichy (@PatTheSilent)
+options:
+  physical_network:
+    description:
+      - Name of the physical network.
+    required: true
+    aliases:
+      - name
+    type: str
+  zone:
+    description:
+      - Name of the zone in which the network belongs.
+      - If not set, default zone is used.
+    type: str
+  broadcast_domain_range:
+    description:
+      - broadcast domain range for the physical network[Pod or Zone].
+    required: false
+    choices: [ POD, ZONE ]
+    type: str
+  domain:
+    description:
+      - Domain the network is owned by.
+    type: str
+  isolation_method:
+    description:
+      - Isolation method for the physical network.
+    required: false
+    choices: [ VLAN, GRE, L3 ]
+    type: str
+  network_speed:
+    description:
+      - The speed for the physical network.
+    required: false
+    choices: [1G, 10G]
+    type: str
+  tags:
+    description:
+      - A tag to identify this network.
+      - Physical networks support only one tag.
+      - To remove an existing tag pass an empty string.
+    required: false
+    aliases:
+      - tag
+    type: str
+  vlan:
+    description:
+      - The VLAN/VNI Ranges of the physical network.
+    required: false
+    type: str
+  nsps_enabled:
+    description:
+      - List of Network Service Providers to enable
+    required: false
+    type: list
+  nsps_disabled:
+    description:
+      - List of Network Service Providers to disable
+    required: false
+    type: list
+  state:
+    description:
+      - State of the physical network.
+    required: false
+    default: present
+    type: str
+    choices: [ present, absent, disabled, enabled ]
+  poll_async:
+    description:
+      - Poll async jobs until job has finished.
+    required: false
+    default: yes
+    type: bool
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+# Ensure a network is present
+- cs_physical_network:
```

Comment:
~~~
- name: Ensure a network is present
  cs_physical_network:
~~~

Reply:
Done.